src/plot-data.ts

```diff
@@ -1932,27 +1932,38 @@ export abstract class PlotData {
 
   refresh_clicked_point_index() {  //selected_clicked_index : index of selected points in the initial point list
     this.clicked_point_index = [];
-    for (let i=0; i<this.clicked_points.length; i++) {
+    for (let i = 0; i < this.clicked_points.length; i++) {
       if (this.clicked_points[i] === undefined) continue;
       if ((this.plotObject['type_'] == 'scatterplot') && this.clicked_points[i]) {
         let true_clicked_points = this.clicked_points[i].points_inside;
-        for (let j=0; j<true_clicked_points.length; j++) {
+        for (let j = 0; j < true_clicked_points.length; j++) {
           this.clicked_point_index.push(List.get_index_of_element(true_clicked_points[j], this.plotObject.point_list));
         }
       } else if (this.plotObject['type_'] == 'graph2D') {
-        for (let j=0; j<this.plotObject.graphs.length; j++) {
+        for (let j = 0; j < this.plotObject.graphs.length; j++) {
           if ((List.is_include(this.clicked_points[i], this.plotObject.graphs[j].point_list)) && this.clicked_points[i]) {
             this.clicked_point_index.push([List.get_index_of_element(this.clicked_points[i], this.plotObject.graphs[j].point_list), j]);
           }
         }
+      } else if ((this.plotObject instanceof PieChart) && this.clicked_points[i]) {
+        let true_clicked_points = this.clicked_points[i].points_inside;
+        for (let j = 0; j < true_clicked_points.length; j++) {
+          this.clicked_point_index.push(List.get_index_of_element(true_clicked_points[j], this.plotObject.pieParts));
+        }
       }
     }
   }
 
 
   refresh_latest_selected_points_index() {
+    let elements_list: Array<any>;
     this.latest_selected_points_index = [];
-    for (let i=0; i<this.latest_selected_points.length; i++) {
+    if (this.plotObject instanceof PieChart) {
+      elements_list = this.plotObject.pieParts;
```

Comment:
Future : To keep in mind while refactoring : In order to avoid these kind of if statements everywhere, as any "plotObject" should inherit from a base class, we should define a getter (js equivalent of a python property), that would enable high customizability when we call the getter as well as guarantee simplicity and genericity.

For ex, here, the base class would define the getter as `return this.point_list` where PieChar would overwrite this behavior to call and yield its own "way" of representing itself, ie. `return this.pieParts`

And the code would only call for the getter